wolfcrypt/src/sha.c

```diff
@@ -212,6 +221,10 @@ int wc_ShaFinal(Sha* sha, byte* hash)
 
 int wc_InitSha(Sha* sha)
 {
+    if (sha == NULL){
+        return BAD_FUNC_ARG;
+    }
+
     int ret = 0;
```

Comment:
Visual studio is very strict about C89. All variables must be declared at the BEGINNING of a statement:

move variable declarations to beginning of statement or change to "declaration then assignement"